realm/realm-library/src/main/cpp/io_realm_internal_Table.cpp

```diff
@@ -271,7 +271,9 @@ JNIEXPORT jint JNICALL Java_io_realm_internal_Table_nativeGetColumnType(JNIEnv*,
     jint column_type = table->get_column_type(column_key);
     if (column_type != type_LinkList &&  table->is_list(column_key)) {
         // add the offset so it can be mapped correctly in Java (RealmFieldType#fromNativeValue)
-        column_type += 128;
+        column_type += 128;     // PropertyType::Array
+    } else if (column_key.is_dictionary()) {
+        column_type += 512;     // PropertyType::Dictionary
```

Comment:
How come that we use 512 instead of 256?

Reply:
The value for a dictionary type is 512, whereas 256 is for set.